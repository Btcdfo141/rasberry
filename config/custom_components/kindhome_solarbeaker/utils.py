def log(msg):
	print(f"MATI: {msg}")

def log(fun, msg):
    print(f"MATI, {fun}: {msg}")
