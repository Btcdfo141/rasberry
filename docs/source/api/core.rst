.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: