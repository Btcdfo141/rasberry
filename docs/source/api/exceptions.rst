.. _exceptions_module:

:mod:`spencerassistant.exceptions`
-------------------------------

.. automodule:: spencerassistant.exceptions
    :members:
