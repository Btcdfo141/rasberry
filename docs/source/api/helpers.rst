:mod:`spencerassistant.helpers`
============================

.. automodule:: spencerassistant.helpers
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.aiohttp\_client
-------------------------------------

.. automodule:: spencerassistant.helpers.aiohttp_client
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.area\_registry
------------------------------------

.. automodule:: spencerassistant.helpers.area_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.check\_config
-----------------------------------

.. automodule:: spencerassistant.helpers.check_config
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.collection
--------------------------------

.. automodule:: spencerassistant.helpers.collection
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.condition
-------------------------------

.. automodule:: spencerassistant.helpers.condition
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_flow
-----------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_oauth2\_flow
-------------------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_oauth2_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_validation
----------------------------------------

.. automodule:: spencerassistant.helpers.config_validation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.data\_entry\_flow
---------------------------------------

.. automodule:: spencerassistant.helpers.data_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.debounce
------------------------------

.. automodule:: spencerassistant.helpers.debounce
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.deprecation
---------------------------------

.. automodule:: spencerassistant.helpers.deprecation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.device\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.device_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.discovery
-------------------------------

.. automodule:: spencerassistant.helpers.discovery
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.dispatcher
--------------------------------

.. automodule:: spencerassistant.helpers.dispatcher
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity
----------------------------

.. automodule:: spencerassistant.helpers.entity
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_component
---------------------------------------

.. automodule:: spencerassistant.helpers.entity_component
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_platform
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_values
------------------------------------

.. automodule:: spencerassistant.helpers.entity_values
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entityfilter
----------------------------------

.. automodule:: spencerassistant.helpers.entityfilter
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.event
---------------------------

.. automodule:: spencerassistant.helpers.event
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.icon
--------------------------

.. automodule:: spencerassistant.helpers.icon
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.integration\_platform
-------------------------------------------

.. automodule:: spencerassistant.helpers.integration_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.intent
----------------------------

.. automodule:: spencerassistant.helpers.intent
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.json
--------------------------

.. automodule:: spencerassistant.helpers.json
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.location
------------------------------

.. automodule:: spencerassistant.helpers.location
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.logging
-----------------------------

.. automodule:: spencerassistant.helpers.logging
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.network
-----------------------------

.. automodule:: spencerassistant.helpers.network
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.restore\_state
------------------------------------

.. automodule:: spencerassistant.helpers.restore_state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.script
----------------------------

.. automodule:: spencerassistant.helpers.script
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.service
-----------------------------

.. automodule:: spencerassistant.helpers.service
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.signal
-----------------------------

.. automodule:: spencerassistant.helpers.signal
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.state
---------------------------

.. automodule:: spencerassistant.helpers.state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.storage
-----------------------------

.. automodule:: spencerassistant.helpers.storage
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.sun
-------------------------

.. automodule:: spencerassistant.helpers.sun
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.system\_info
----------------------------------

.. automodule:: spencerassistant.helpers.system_info
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.temperature
---------------------------------

.. automodule:: spencerassistant.helpers.temperature
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.template
------------------------------

.. automodule:: spencerassistant.helpers.template
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.translation
---------------------------------

.. automodule:: spencerassistant.helpers.translation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.typing
----------------------------

.. automodule:: spencerassistant.helpers.typing
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.update\_coordinator
-----------------------------------------

.. automodule:: spencerassistant.helpers.update_coordinator
   :members:
   :undoc-members:
   :show-inheritance:
