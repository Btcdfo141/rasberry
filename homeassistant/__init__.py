"""Init file for Home Assistant."""


test2
