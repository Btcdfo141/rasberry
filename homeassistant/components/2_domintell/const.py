"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
