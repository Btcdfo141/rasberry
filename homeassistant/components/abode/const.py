"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
