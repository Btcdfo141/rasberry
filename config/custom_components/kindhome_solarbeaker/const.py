from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
