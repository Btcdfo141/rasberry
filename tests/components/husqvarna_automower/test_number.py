"""Tests for number platform."""

from unittest.mock import AsyncMock, patch

from aioautomower.exceptions import ApiException
from aioautomower.utils import mower_list_to_dictionary_dataclass
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.husqvarna_automower.const import DOMAIN
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import HomeAssistantError
from homeassistant.helpers import entity_registry as er

from . import setup_integration
from .const import TEST_MOWER_ID

from tests.common import MockConfigEntry, load_json_value_fixture, snapshot_platform


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_number_commands(
    hass: HomeAssistant,
    mock_automower_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test number commands."""
    entity_id = "number.test_mower_1_cutting_height"
    await setup_integration(hass, mock_config_entry)
    await hass.services.async_call(
        domain="number",
        service="set_value",
        target={"entity_id": entity_id},
        service_data={"value": "3"},
        blocking=True,
    )
    mocked_method = mock_automower_client.set_cutting_height
    assert len(mocked_method.mock_calls) == 1

    mocked_method.side_effect = ApiException("Test error")
    with pytest.raises(HomeAssistantError) as exc_info:
        await hass.services.async_call(
            domain="number",
            service="set_value",
            target={"entity_id": entity_id},
            service_data={"value": "3"},
            blocking=True,
        )
    assert (
        str(exc_info.value)
        == "Command couldn't be sent to the command queue: Test error"
    )
    assert len(mocked_method.mock_calls) == 2


async def test_number_workarea_commands(
    hass: HomeAssistant,
    mock_automower_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test number commands."""
    entity_id = "number.test_mower_1_cutting_height_front_lawn"
    await setup_integration(hass, mock_config_entry)
    values = mower_list_to_dictionary_dataclass(
        load_json_value_fixture("mower.json", DOMAIN)
    )
    values[TEST_MOWER_ID].work_areas[123456].cutting_height = 75
    mock_automower_client.get_status.return_value = values
    await hass.services.async_call(
        domain="number",
        service="set_value",
        target={"entity_id": entity_id},
        service_data={"value": "75"},
        blocking=True,
    )
    mocked_method = mock_automower_client.set_cutting_height_workarea
    assert len(mocked_method.mock_calls) == 1
    state = hass.states.get(entity_id)
    assert state.state is not None
    assert state.state == "75"

    mocked_method.side_effect = ApiException("Test error")
    with pytest.raises(HomeAssistantError) as exc_info:
        await hass.services.async_call(
            domain="number",
            service="set_value",
            target={"entity_id": entity_id},
            service_data={"value": "75"},
            blocking=True,
        )
    assert (
        str(exc_info.value)
        == "Command couldn't be sent to the command queue: Test error"
    )
    assert len(mocked_method.mock_calls) == 2


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_snapshot_number(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    mock_automower_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test states of the number entity."""
    with patch(
        "homeassistant.components.husqvarna_automower.PLATFORMS",
        [Platform.NUMBER],
    ):
        await setup_integration(hass, mock_config_entry)
        await snapshot_platform(
            hass, entity_registry, snapshot, mock_config_entry.entry_id
        )
