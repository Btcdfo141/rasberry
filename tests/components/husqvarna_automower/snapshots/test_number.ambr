# serializer version: 1
# name: test_snapshot_number[number.test_mower_1_cutting_height-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 9,
      'min': 1,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'number.test_mower_1_cutting_height',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Cutting height',
    'platform': 'husqvarna_automower',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'cutting_height',
    'unique_id': 'c7233734-b219-4287-a173-08e3643f89f0_cutting_height',
    'unit_of_measurement': None,
  })
# ---
# name: test_snapshot_number[number.test_mower_1_cutting_height-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Mower 1 Cutting height',
      'max': 9,
      'min': 1,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'context': <ANY>,
    'entity_id': 'number.test_mower_1_cutting_height',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '4',
  })
# ---
# name: test_snapshot_number[number.test_mower_1_cutting_height_back_lawn-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 100,
      'min': 0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'number.test_mower_1_cutting_height_back_lawn',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Cutting height Back lawn',
    'platform': 'husqvarna_automower',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'cutting_height_work_area',
    'unique_id': 'c7233734-b219-4287-a173-08e3643f89f0_cutting_height_work_area_654321',
    'unit_of_measurement': None,
  })
# ---
# name: test_snapshot_number[number.test_mower_1_cutting_height_back_lawn-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Mower 1 Cutting height Back lawn',
      'max': 100,
      'min': 0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'context': <ANY>,
    'entity_id': 'number.test_mower_1_cutting_height_back_lawn',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '25',
  })
# ---
# name: test_snapshot_number[number.test_mower_1_cutting_height_front_lawn-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 100,
      'min': 0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'number.test_mower_1_cutting_height_front_lawn',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Cutting height Front lawn',
    'platform': 'husqvarna_automower',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'cutting_height_work_area',
    'unique_id': 'c7233734-b219-4287-a173-08e3643f89f0_cutting_height_work_area_123456',
    'unit_of_measurement': None,
  })
# ---
# name: test_snapshot_number[number.test_mower_1_cutting_height_front_lawn-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Mower 1 Cutting height Front lawn',
      'max': 100,
      'min': 0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'context': <ANY>,
    'entity_id': 'number.test_mower_1_cutting_height_front_lawn',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '50',
  })
# ---
# name: test_snapshot_number[number.test_mower_1_cutting_height_none-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 100,
      'min': 0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'number.test_mower_1_cutting_height_none',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Cutting height None',
    'platform': 'husqvarna_automower',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'cutting_height_work_area',
    'unique_id': 'c7233734-b219-4287-a173-08e3643f89f0_cutting_height_work_area_0',
    'unit_of_measurement': None,
  })
# ---
# name: test_snapshot_number[number.test_mower_1_cutting_height_none-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Mower 1 Cutting height None',
      'max': 100,
      'min': 0,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 1.0,
    }),
    'context': <ANY>,
    'entity_id': 'number.test_mower_1_cutting_height_none',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '50',
  })
# ---
